client/service/src/builder.rs

```diff
@@ -1495,3 +1241,384 @@ ServiceBuilder<
+)
+	where
+		TBl: BlockT,
+		TExec: CallExecutor<TBl> + 'static,
+		TRtApi: 'static,
+		TBackend: sc_client_api::backend::Backend<TBl> + 'static,
+		Client<TBackend, TExec, TBl, TRtApi>: ProvideRuntimeApi<TBl> + Send + Sync,
+		<Client<TBackend, TExec, TBl, TRtApi> as ProvideRuntimeApi<TBl>>::Api:
+			sc_offchain::OffchainWorkerApi<TBl>
+{
+	client.import_notification_stream().for_each(move |n| {		
+		if n.is_new_best {
+			notifications_spawn_handle.spawn(
+				"offchain-on-block",
+				offchain.on_block_imported(
+					&n.header,
+					network_state_info.clone(),
+					is_validator,
+				),
+			);
+		} else {
+			log::debug!(
+				target: "sc_offchain",
+				"Skipping offchain workers for non-canon block: {:?}",
+				n.header,
+			)
+		}
+
+		ready(())
+	}).await;
+}
+
+async fn extrinsic_notifications<TBl, TExPool>(
+	transaction_pool: Arc<TExPool>,
+	network: Arc<NetworkService<TBl, <TBl as BlockT>::Hash>>
+)
+	where
+		TBl: BlockT,
+		TExPool: MaintainedTransactionPool<Block=TBl, Hash = <TBl as BlockT>::Hash>,
+{
+	// extrinsic notifications
+	transaction_pool.import_notification_stream()
+		.for_each(move |hash| {
+			network.propagate_extrinsic(hash);
+			let status = transaction_pool.status();
+			telemetry!(SUBSTRATE_INFO; "txpool.import";
+				"ready" => status.ready,
+				"future" => status.future
+			);
+			ready(())
+		})
+		.await;
+}
+
+// Periodically notify the telemetry.
+async fn telemetry_periodic_send<TBl, TBackend, TExec, TRtApi, TExPool>(
+	client: Arc<Client<TBackend, TExec, TBl, TRtApi>>,
+	transaction_pool: Arc<TExPool>,
+	mut metrics_service: MetricsService,
+	network_status_sinks: Arc<Mutex<status_sinks::StatusSinks<(NetworkStatus<TBl>, NetworkState)>>>
+)
+	where
+		TBl: BlockT,
+		TExec: CallExecutor<TBl>,
+		Client<TBackend, TExec, TBl, TRtApi>: ProvideRuntimeApi<TBl>,
+		TExPool: MaintainedTransactionPool<Block=TBl, Hash = <TBl as BlockT>::Hash>,
+		TBackend: sc_client_api::backend::Backend<TBl>,
+{
+	let (state_tx, state_rx) = tracing_unbounded::<(NetworkStatus<_>, NetworkState)>("mpsc_netstat1");
+	network_status_sinks.lock().push(std::time::Duration::from_millis(5000), state_tx);
+	state_rx.for_each(move |(net_status, _)| {
+		let info = client.usage_info();
+		metrics_service.tick(
+			&info,
+			&transaction_pool.status(),
+			&net_status,
+		);
+		ready(())
+	}).await;
+}
+
+async fn telemetry_periodic_network_state<TBl: BlockT>(
+	network_status_sinks: Arc<Mutex<status_sinks::StatusSinks<(NetworkStatus<TBl>, NetworkState)>>>
+) {
+	// Periodically send the network state to the telemetry.
+	let (netstat_tx, netstat_rx) = tracing_unbounded::<(NetworkStatus<_>, NetworkState)>("mpsc_netstat2");
+	network_status_sinks.lock().push(std::time::Duration::from_secs(30), netstat_tx);
+	netstat_rx.for_each(move |(_, network_state)| {
+		telemetry!(
+			SUBSTRATE_INFO;
+			"system.network_state";
+			"state" => network_state,
+		);
+		ready(())
+	}).await;
+}
+
+async fn informant<TBl, TBackend, TExec, TRtApi, TExPool>(
```

Comment:
What is the reason of this function? It just wraps another function call?

Reply:
it was to simplify the function call, but I think I can just move those changes into `sc_informant::build` itself.